sklearn/preprocessing/_data.py

```diff
@@ -471,22 +475,24 @@ def partial_fit(self, X, y=None):
                 "Consider using MaxAbsScaler instead."
             )
 
+        xp, _ = get_namespace(X)
+
         first_pass = not hasattr(self, "n_samples_seen_")
         X = self._validate_data(
             X,
             reset=first_pass,
-            dtype=FLOAT_DTYPES,
+            dtype=(xp.float64, xp.float32, xp.float16),
```

Comment:
What shall we do about the fact that `float16` is not part of the Array API? I assume it is important to have it here (even if we did remove it, it would mean a deprecation cycle). Should we add it to a scikit-learn internal "compat layer"? `float16` seems to exist in Numpy, Torch and cupy.

Thoughts?

Reply:
For torch (not checked cupy) `xp.float16` works/exists. I think as part of the Array API standard it is allowed to add additional dtypes (in this case float16), but `numpy.array_api` only contains the required types.